docs/pages/router/reference/testing.mdx

```diff
@@ -0,0 +1,100 @@
+---
+title: Testing
+description: Learn how to use create integration tests for your app.
+---
+
+Expo Router relies on your file system, which can present challenges when setting up mocks for integration tests. Expo Router's submodule, `expo-router/testing-library`, is a set of testing utilities built on top of the popular [`@testing-library/react-native`](https://callstack.github.io/react-native-testing-library/) and allows you to quick create minimal in-memory Expo Router apps, pre-configured for testing.
+
+## Configuration
+
+Before you proceed, ensure you've set up `jest-expo` according to the [Unit Testing documentation](/develop/testing) and [`@testing-library/react-native`](https://callstack.github.io/react-native-testing-library/docs/getting-started).
+
+## renderRouter
+
+`renderRouter` extends the functionality of [`render`](https://callstack.github.io/react-native-testing-library/docs/api#render) to simplify testing with Expo Router. It returns the the same query object as [`render`](https://callstack.github.io/react-native-testing-library/docs/api#render), and is compatible with [`screen`](https://callstack.github.io/react-native-testing-library/docs/api#screen), allowing you to use the standard [query API](https://callstack.github.io/react-native-testing-library/docs/api-queries) to locate components.
+
+`renderRouter` accepts the same [options](https://callstack.github.io/react-native-testing-library/docs/api#render-options) as `render` and introduces an additional option `initialRoute`, which sets an initial route for simulating deep-linking.
+
+### Inline file-system mock
+
+`renderRouter(mock: Record<string, ReactComponent>, options: RenderOptions)`
+
+`renderRouter` can provide inline-mocking of a file system by passing an object to this function as the first parameter. The keys of the object are the mock filesystem paths. Importantly, when defining these paths, do not use leading relative (`./`) or absolute (`/`) notation and exclude file extension.
```

Comment:
```suggestion
`renderRouter` can provide inline-mocking of a file system by passing an object to this function as the first parameter. The keys of the object are the mock filesystem paths. **Do not use leading relative (`./`) or absolute (`/`) notation when defining these paths and exclude file extension.**
```

We can highlight the context here if it is important for the developer to pay attention.